app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  before_action :authorize_user
+
+  def show
+    @user = User.new
+    @users = User.order(:username)
+
+    if params[:user] != nil
+      username = params[:user][:username]
+      @user = User.find_by(username: username)
+    end
+  end
+
+  def index
```

Comment:
here, it would be expected from a RESTful convention standpoint that we would be dealing with a list of users. Why are we showing only one user here?

Also, if this is what we want functionally, we should defensively check to see if `params[:user]` is in fact a hash for this conditional.

```
if params[:user] && params[:user][:id]
```

Reply:
We removed the index controller as it wasn't needed.